docs/cmake-presets.md

```diff
@@ -0,0 +1,272 @@
+* **CMake: Select Variant**
+* **CMake: Scan for Kits**
+* **CMake: Select a Kit**
+* **CMake: Edit User-Local CMake Kits**
+
+## Troubleshooting
+
+If things aren’t working as expected, there are a few troubleshooting steps that you can take.
+
+If either `CMakePresets.json` or `CMakeUserPresets.json` is invalid, then none of the presets in the invalid file will be available for selection. CMake Tools IntelliSense can help you catch many of these JSON errors, but it won’t know if you're referencing a preset with `inherits` or `configurePreset` by the wrong name. To check if your preset files are valid, run `cmake --list-presets` from the command line at the root of your project directory (CMake 3.20 or higher is required). If either file is invalid, then you'll see the following error:
+
+```DOS
+CMake Error: Could not read presets from
+C:/Users/<user>/source/repos/<project-name>: JSON parse error
+```
+
+If you're working on Windows then CMake must be on the `PATH`.
+
+Other troubleshooting steps include:
+* Confirm cmake.exe and your generator are installed and on the `PATH`.
+* Delete the cache and reconfigure the project (**CMake: Delete Ca* and Reconfigure**)
+* Reset the CMake Tools extension state (**CMake: Reset CMake Tool* Extension State**)
+* Increase the logging level (`cmake.loggingLevel` in `settings.json`) and/or check the log file (**CMake: Open the CMake Tools Log File**)
+
+If you have identified a problem, the best way to report it is by submitting an issue to the [CMake Tools extension repository](https://github.com/microsoft/vscode-cmake-tools).
+
+## Run CMake from the command line or a continuous integration (CI) pipeline
+
+You can use the same `CMakePresets.json` and `CMakeUserPresets.json` files to invoke CMake in Visual Studio Code, and from the command line. The [CMake](https://cmake.org/cmake/help/latest/manual/cmake.1.html) and [CTest](https://cmake.org/cmake/help/latest/manual/ctest.1.html) documentation are the best resources for invoking CMake and CTest with `--preset`. CMake version 3.20 or higher is required.
+
+The following commands can be run from the directory where your `CMakePresets.json` is located:
+
+```DOS
+cmake --list-presets=all .
+cmake --preset <configurePreset-name>
+cmake --build --preset <buildPreset-name> 
+ctest --preset <testPreset-name>
+```
+
+## Sourcing the environment when building with command-line generators on Windows
+
+See [Sourcing the environment when building with command-line generators on Windows](https://docs.microsoft.com/cpp/build/cmake-presets-vs#sourcing-the-environment-when-building-with-command-line-generators-on-windows) documentation for more information on building with a command-line generator on Windows.
+
+## Example `CMakePresets.json` file
+
+See the `CMakePresets.json` file checked in the [box2d-lite](https://github.com/esweet431/box2d-lite/blob/main/`CMakePresets.json`) code sample. It contains examples of Configure Presets, Build Presets, and Test Presets.
```

Comment:
This link isn't working for some reason - try this? https://github.com/esweet431/box2d-lite/

Reply:
I think what you want is https://github.com/esweet431/box2d-lite/blob/main/CMakePresets.json